### ISSUE TITLE
custom default_file_mode ignored for remote connections on iRES.

### ISSUE THREAD

**Issue by kript** [irods/irods#4124]
### iRODS Version, OS and Version

4.1.10
4.1.12
RHEL 7.4
Ubuntu 12.04

### What did you try to do?

Upload a file where the `default_file_mode` was set  _only on the iRES_  to `0640` (i.e. group read/execute).

### Expected behavior

File would be written to the local filesystem resource with 0640 permissions.

### Observed behavior (including steps to reproduce, if applicable)

If file was uploaded locally to the iRES (say by using the local irods user to test), then the **local** `/etc/irods/server_config.json` configuration was honoured. 

If the file was uploaded via the IES, then the value of `default_file_mode` in `/etc/irods/server_config.json` **from the IES** was used instead, overwriting that of the local iRES.

This is not clear this is intended from the documentation, and not preferred in our use case (we would like the ability to set the umask per server, if needed).

**Comment by trel** [comment]
Agreed - I would expect the server doing the writing, rather than the server serving the connection request to be the one that is honored.
